cmd/manager/scap.go

```diff
@@ -311,6 +323,34 @@ func fetch(client *kubernetes.Clientset, objects []utils.ResourcePath) (map[stri
 	return results, warnings, nil
 }
 
+func filter(ctx context.Context, rawobj []byte, filter string) ([]byte, error) {
+	fltr, fltrErr := gojq.Parse(filter)
+	if fltrErr != nil {
+		return nil, fmt.Errorf("could not create filter '%s': %w", filter, fltrErr)
+	}
+	obj := map[string]interface{}{}
+	unmarshallErr := json.Unmarshal(rawobj, &obj)
+	if unmarshallErr != nil {
+		return nil, fmt.Errorf("Error unmarshalling json: %w", unmarshallErr)
+	}
+	iter := fltr.RunWithContext(ctx, obj)
+	v, ok := iter.Next()
```

Comment:
So what about the case when the filter would return multiple values? Should we try running iter.Next() and fail if the result is not a single object? Because right now if the filter yields multiple objects we only return the first one (multiple objects is AFAIK something else than an array..)

Reply:
That's right, we have the limitation that we can only support one output object and that's deliberate. Multiple objects won't lead to a parseable YAML file, and thus are quite useless to us in that state.

Reply:
Right, that was my question, should we run another get from the iterator to assert that we only get one result? But feel free to add this in another PR or not at all of you disagree.